daemon/docker_watcher.go

```diff
@@ -258,35 +258,30 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
 			continue
 		}
 
-		var orchLabelsModified bool
 		d.containersMU.RLock()
 		cont, ok := d.containers[id]
 		d.containersMU.RUnlock()
-		if ok {
-			cont.Mutex.Lock()
-			if orchLabelsModified = updateOrchLabels(cont, lbls); !orchLabelsModified {
```

Comment:
@aanm this is what I referred to as the original logic for skipping if labels modified. To skip is should satisfy both ok on line 265 and ! orchLabelsModified.